src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -238,28 +249,132 @@ private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, ob
             // TCP Format: 
             // tcp:<host name>\<instance name>
             // tcp:<host name>,<TCP/IP port number>
-            int portNumber = 1433;
-            string[] serverAndPortParts = fullServerName.Split(',');
 
-            if (serverAndPortParts.Length == 2)
+            if (String.IsNullOrWhiteSpace(fullServerName))
+            {
+                return new SNITCPHandle("localhost", DefaultSqlServerPort, timerExpire, callbackObject, parallel);
+            }
+
+            string[] serverNamePartsByComma = fullServerName.Split(s_commaSeparator);
+            string[] serverNamePartsByBackSlash = fullServerName.Split(s_backSlashSeparator);
+            if (serverNamePartsByComma.Length < 2 && serverNamePartsByBackSlash.Length < 2)
             {
+                return new SNITCPHandle(fullServerName, DefaultSqlServerPort, timerExpire, callbackObject, parallel);
+            }
+            else if (serverNamePartsByComma.Length == 2 && serverNamePartsByBackSlash.Length < 2)
+            {
+                string hostName = serverNamePartsByComma[0];
+                int portNumber = -1;
                 try
                 {
-                    portNumber = ushort.Parse(serverAndPortParts[1]);
+                    portNumber = ushort.Parse(serverNamePartsByComma[1]);
```

Comment:
I think our intention is not to swallow the exception by `TryParse()` but to propagate to the surface by creating SNIError object which can contain the parse exception.